### PR TITLE
modify pom to run make clean in installer directory

### PR DESCRIPTION
**Pull request by pameyer**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #3974 : mvn clean doesn't clean installer directory

developer only change

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None 
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/4.6.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

**Comment by pdurbin** [comment]
@pameyer this pull request has merge conflicts. Let's figure out what to do with it and if you're still interested in #3974.

**Comment by pameyer** [comment]
@pdurbin I'm still interesting in reducing the number manual steps that can go wrong when building a new installer.  But if I'm remembering correctly there was something you were unhappy about with this approach.

**Comment by pdurbin** [comment]
@pameyer make sense. I think I'd be happier if we also documented the right way to use the addition to the pom.xml at http://guides.dataverse.org/en/4.8.2/developers/making-releases.html . Right now that page simply says "Upload the following artifacts to the draft release you created" but doesn't say how the war file or installer were created. Maybe the first step would be to document our current process. Then we'd see the gap, the problem that you're fixing.

File: pom.xml [hunk 1]
```diff
@@ -576,47 +576,25 @@
...
-                            <fileset>
-                                <directory>target/dataverse-${project.version}.war/WEB-INF/lib/aws-java-sdk-bundle-${aws.version}.jar/META-INF</directory>
-                                <includes>
-                                    <include>javamail.providers</include>
-                                </includes>
-                            </fileset>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
```
Comment (pdurbin) [inline pom.xml]: What is going on with all the stuff being removed above?

File: pom.xml [hunk 2]
```diff
@@ -576,47 +576,25 @@
...
+			<execution>
+				<id>run make clean in installer directory</id>
+				<phase>clean</phase>
+				<goals>
+					<goal>exec</goal>
+				</goals>
+				<configuration>
+					<executable>make</executable>
+					<commandlineArgs>clean</commandlineArgs>
+					<workingDirectory>scripts/installer</workingDirectory>
```
Comment (pdurbin) [inline pom.xml]: I guess this part seem safe enough to merge. I do run `mvn clean` quite a bit and it feels like action at a distance to have it mess around with what I consider to be a different build process in a different directory (`scripts/installer`) but in practice, I don't see the harm in more cleanup happening. We could always revert this pull request if it starts causing us problems.

**Comment by pdurbin** [comment]
@pameyer and I talked out this pull request. We're both a little confused by the deletion portion but figure it has to do with a recent merge from develop. The substance of the pull request is in the orginal commit at 3a6c89e but to me it would be surprising behavior (action at a distance) to have Maven clear out a directory somewhere else on the file system. Basically, we have one build tool for dealing with the Java code, which is Maven, and another tool to create the dvinstall.zip file, which is Make. I like the separation. I don't particularly want these tools to be entangled so I'm closing this pull request.

@pameyer has very helpfully created #4528 which expresses the problem he's facing (`make` doesn't know the war file has been updated) so let's focus our energies there.

**Pull request closed without merging**